src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy

```diff
@@ -331,6 +317,24 @@ type Query {
 
     }
 
+    def "prints directives"() {
+        given:
+        GraphQLFieldDefinition fieldDefinition = newFieldDefinition()
+                .name("field")
+                .deprecate("reason")
+                .type(GraphQLString).build()
+        def queryType = GraphQLObjectType.newObject().name("Query").field(fieldDefinition).build()
+        def schema = GraphQLSchema.newSchema().query(queryType).build()
+        when:
+        def result = new SchemaPrinter().print(schema)
+
+        then:
+        result == """type Query {
+  field: String @deprecate(reason: "reason")
+}
+"""
```

Comment:
Yeah the test coverage here is not wide enough.  It does not cover the 7 different types of objects